sdk/scala-sdk/src/main/scala/com/akkaserverless/scalasdk/impl/view/ViewAdapter.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.akkaserverless.scalasdk.impl.view
+
+import java.util.Optional
+
+import scala.jdk.OptionConverters._
+
+import com.akkaserverless.javasdk
+import com.akkaserverless.javasdk.ServiceCallFactory
+import com.akkaserverless.scalasdk.Metadata
+import com.akkaserverless.scalasdk.impl.MetadataImpl
+import com.akkaserverless.scalasdk.view.UpdateContext
+import com.akkaserverless.scalasdk.view.View
+import com.akkaserverless.scalasdk.view.ViewCreationContext
+import com.akkaserverless.scalasdk.view.ViewProvider
+import com.google.protobuf.Descriptors
+
+private[scalasdk] class Scala2JavaViewAdapter[S](scalasdkView: View[S]) extends javasdk.view.View[S] {
```

Comment:
This adapter approach is completely untested and I'm not sure it's the right approach.

Reply:
This is the "flow":

1. AkkaServerless.register(scalasdk.ViewProvider)
  - delegates to javasdk by wrapping the provider with Scala2JavaViewProviderAdapter
2. javasdk.ViewProvider.newHandler is invoked
  - adapter delegates to scaladsl new handler and adapts the ViewCreationContext
  - the action View implementation is created
3. For each incoming event
  - javasdk.ViewHandler._internalSetUpdateContext  is invoked
    - adapter delegates to scaladsl and adapts the UpdateContext
  - javasdk.ViewHandler.handleUpdate is invoked
    - adapter delegates to scaladsl
    - adapter converts the returned effect

Reply:
A bit of wrapping/unwrapping work but sounds reasonable.